crates/solidity/testing/perf/npm-benches/src/main.rs

```diff
@@ -0,0 +1,192 @@
+        fetch(hash, input_path)?;
+        let path = input_path.join(hash);
+
+        let mut results = vec![];
+        for sut in SubjectUT::iter() {
+            let mut sut_result = self.run(&path, file, sut)?;
+            results.append(&mut sut_result);
+        }
+
+        Ok(results)
+    }
+
+    fn compute_regex(&self) -> Result<regex::Regex> {
+        if let Some(pattern) = &self.pattern {
+            Ok(regex::Regex::new(pattern)?)
+        } else {
+            Ok(regex::Regex::new(".*")?)
+        }
+    }
+
+    fn individual_file_benchmarks(
+        &self,
+        input_path: &Path,
+        files: &Vec<File>,
+    ) -> Result<Vec<Timing>> {
+        let mut results = vec![];
+
+        let pattern_regex = self.compute_regex()?;
+        for file in files {
+            if !pattern_regex.is_match(&file.file) {
+                continue;
+            }
+
+            let mut result = self.run_benchmarks(input_path, &file.hash, Some(&file.file))?;
+            results.append(&mut result);
+        }
+        Ok(results)
+    }
+
+    fn project_benchmarks(
+        &self,
+        input_path: &Path,
+        projects: &Vec<Project>,
+    ) -> Result<Vec<Timing>> {
+        let mut results = vec![];
+        let pattern_regex = self.compute_regex()?;
+
+        for project in projects {
+            if !pattern_regex.is_match(&project.name) {
+                continue;
+            }
+
+            let mut result = self.run_benchmarks(input_path, &project.hash, None)?;
+            results.append(&mut result);
+        }
+        Ok(results)
+    }
+
+    fn run(
+        &self,
+        path: &Path,
+        file: Option<&str>,
+        sut: SubjectUT,
+    ) -> Result<Vec<Timing>, anyhow::Error> {
+        let perf_crate = CargoWorkspace::locate_source_crate("solidity_testing_perf")?;
+        let mut command = Command::new("npx")
+            .arg("tsx")
+            .flag("--trace-uncaught")
+            .flag("--expose-gc")
+            .arg(perf_crate.join("npm/src/main.mts").to_str().unwrap())
+            .property("--dir", path.to_string_lossy());
+
+        if let Some(file) = file {
+            command = command.property("--file", file);
+        }
+
+        command = command
+            .property("--runner", sut.as_ref())
+            .property("--cold", self.cold.to_string())
+            .property("--hot", self.hot.to_string());
+
+        let result = command.evaluate()?;
+
+        match serde_json::from_str(&result) {
+            Ok(output) => Ok(output),
+            Err(e) => {
+                eprintln!("Error: Can't parse output as json.\n\tOutput: {result}");
+                Err(e.into())
+            }
+        }
+    }
+}
+
+fn publish<'a>(results: impl Iterator<Item = &'a Timing>) -> Result<()> {
+    let mut output = std::io::stdout();
+    writeln!(output, "{{")?;
+    writeln!(output, "\t\"npm_benchmarks\": {{")?;
+
+    let mut buffer = None;
+    for timing in results {
```

Comment:
I wonder if this is setting the "measure" name to the test name, rather than the actual measure (`Duration` in this case)?